pkgs/node/src/node.zig

```diff
@@ -155,7 +157,34 @@ pub const BeamNode = struct {
             },
         }
 
-        try self.chain.onGossip(data, sender_peer_id);
+        const result = try self.chain.onGossip(data, sender_peer_id);
+        self.handleGossipProcessingResult(result);
+    }
+
+    fn handleGossipProcessingResult(self: *Self, result: chainFactory.GossipProcessingResult) void {
+        // Process successfully imported blocks to retry any cached descendants
+        if (result.processed_block_root) |processed_root| {
+            self.logger.debug(
+                "Gossip block 0x{s} successfully processed, checking for cached descendants",
+                .{std.fmt.fmtSliceHexLower(processed_root[0..])},
+            );
+            self.processCachedDescendants(processed_root);
```

Comment:
if a block was rejected because it was before finalized, we also need to prune out all its descendants

Reply:
in that case we also ignore missing attestation roots